src/state.rs

```diff
@@ -52,79 +123,78 @@ impl NetworkState {
         Self::default()
     }
 
+    /// Returns the state of contracts in the `head`.
+    pub(crate) fn head_mut(&mut self) -> &mut Contracts {
+        &mut self.head
+    }
+
+    /// Returns a reference to the specified contracts state in the `head`
+    /// state.
+    pub fn get_contract<'a>(
+        &'a self,
+        contract_id: &ContractId,
+    ) -> Result<impl Deref<Target = Contract> + 'a, VMError> {
+        self.head.get_contract(contract_id)
+    }
+
+    /// Returns a mutable reference to the specified contracts state in the
+    /// `origin` state.
+    pub fn get_contract_mut<'a>(
+        &'a mut self,
+        contract_id: &ContractId,
+    ) -> Result<impl DerefMut<Target = Contract> + 'a, VMError> {
+        self.head.get_contract_mut(contract_id)
+    }
+
+    /// Get a reference to a particular module from the store.
+    pub fn get_module_ref<'a>(
+        &'a self,
+        id: &'a ContractId,
+    ) -> HostModuleRef<'a> {
+        self.modules.get_module_ref(id)
+    }
```

Comment:
This is not related to the rollbacks' change.

Reply:
Ditto